code/game/machinery/_machinery.dm

```diff
@@ -361,6 +362,10 @@
 /obj/machinery/proc/dump_inventory_contents(list/subset = null)
 	var/turf/this_turf = get_turf(src)
 	for(var/atom/movable/movable_atom in contents)
+		//so machines like microwaves dont dump out signalers after cooking
+		if(istype(movable_atom, /obj/item/assembly/signaler))
```

Comment:
We are close to it, but cuh implementation means ANY signaler won't be dumped at all. Try this and tell me if it works:
```suggestion
		//so machines like microwaves dont dump out signalers after cooking
		if(wires && movable_atom in flatten_list(wires.assemblies))
```